olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -431,13 +472,19 @@ DefaultCacheImpl::GetFromDiscCache(const std::string& key) {
   if (mutable_cache_) {
     auto expiry = GetRemainingExpiryTime(key, *mutable_cache_);
     if (expiry <= 0) {
-      PurgeDiskItem(key, *mutable_cache_);
-      if (mutable_cache_lru_) {
-        mutable_cache_lru_->Erase(key);
+      const auto result = mutable_cache_->Get(key);
+      if (result) {
+        mutable_cache_data_size_ -= key.size() + result.value().size();
       }
+      mutable_cache_data_size_ -= GetExpirySize(key, *mutable_cache_);
```

Comment:
As mentioned above this is not correct as you subtract the size of the expiration key even if we do not have any expiration key in cache